### ISSUE TITLE
Built-In CSS-Support failing for valid CSS

### ISSUE THREAD

**Issue by MrAvantiC** [zeit/next.js#12207]
Hello there!

# Bug report

Built-In CSS-Support failing for valid CSS.

## Describe the bug

When attempting to import a CSS file using the built-in css-imports fails. 
I was able to pinpoint the exact rule that causes the complication, see below.

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Clone this repository: https://github.com/MrAvantiC/next-css-import-issue
2. Run `npm install && npm run dev`
3. You will notice that the build process will fail with the following line:
```sh
[ error ] ./public/main.css (./node_modules/css-loader/dist/cjs.js??ref--5-oneOf-5-1!./node_modules/next/dist/compiled/postcss-loader??__nextjs_postcss!./public/main.css)
CssSyntax error: Unclosed bracket (2:2)

  1 | .foo {
> 2 |   flex: 1 0 calc(30% - (30px / 3));
    |  ^
  3 | }
  4 |
```

The CSS itself is completely valid, which i confirmed with several linterns.

The error seems to be caused by the inner bracket inside the `calc` operator because as soon as you rewrite it, e.g. `flex: 1 0 calc(30% - 30px);` everything is working fine.

**Side note**

I tested the exact same setup using the [next-css package](https://github.com/zeit/next-plugins/tree/master/packages/next-css) and *it works completely fine*.
You can find the example in the [with-next-css branch of the same repository](https://github.com/MrAvantiC/next-css-import-issue/tree/with-next-css).

So there seems to be a difference between both mechanisms.

## Expected behavior

The server should compile without problems as it does using the next-css package, see above.

## System information

- OS: macOs
- Version of Next.js:9.3.5
- Version of Node.js: 13.9.0

**Comment by mattcarlotta** [comment]
Several options:

1.) Move the CSS file out of the `public` folder. See [with-next-css](https://github.com/zeit/next.js/tree/canary/examples/with-next-css) example.

2.) If you want to keep it within the `public` folder, then you'll need a custom `_app.js` file that utilizes `Head` to inject a CSS style `link`. See this [with-public-css](https://github.com/mattcarlotta/with-public-css) example.

**Comment by MrAvantiC** [comment]
Hi @mattcarlotta,
thanks for the answer.

I'm not at the computer at the moment to try it out but I assume you have.

Can you explain why my approach won't work?
Or more exactly: Why *does* the general approach work but it fails for this exact line of CSS? What is the difference?

**Comment by mattcarlotta** [comment]
TLDR: Assets placed in the `public` folder are considered external resources and **shouldn't** be directly imported.

You're trying to import from the `public` folder which is not supported by `next` (not sure about `next-css`). By moving it out of the public folder and into the root (or any other user-defined folder), it'll be properly handled by next's internal Webpack's `css-loader` plugin during compilation.

Any items placed within the `public` folder should be considered external [static file](https://nextjs.org/docs/basic-features/static-file-serving) assets. Static assets are requested/retrieved during the page initialization. It makes a request to your server and retrieves the external asset -- in other words, it's not bundled nor imported/required by a single page, instead, it's an external resource that's being required by all pages/components (like a `favicon`).

For example, you **shouldn't** be importing a static CSS file like so:
```js
import "../public/example.css"
```
Instead, either move it into another folder and import it into `_app.js` as a [global stylesheet](https://nextjs.org/docs/basic-features/built-in-css-support#adding-a-global-stylesheet):
```js
import "../styles/example.css"
```
or place it within the `head` and reference it with the base `/` URL:
```js
<Head>
  <link rel="stylesheet" href="/example.css" />
</Head>
```

When in production, any static assets within the `public` folder will be moved to your compiled folder and be accessible by to the client by using the base `/` path:
```
<link rel="stylesheet" href="/example.css" />
<img src="/example.png" />
...etc
```

That said if you don't want to make a request for an external CSS asset, and the CSS is considered global (required by all files), then consider using Internal CSS with the `style` tag -- which attaches directly to the `html` file and skips requesting an external CSS resource. Read more [here](https://www.w3schools.com/css/css_howto.asp).

**Comment by mattcarlotta** [comment]
@MrAvantiC Apologies. Looks like I jumped the gun on this issue thinking next was not handling css imports from within public. There appears to be a problem with the `postcss-loader` configuration. Attempting to track down which is causing the issue.

**Comment by mattcarlotta** [comment]
Update: The issue appears to be coming from this package: `postcss-flexbug-fixes`. Removing it fixes the error. Also, the reason `next-css` doesn't crash is because it has an empty `postcss-loader` config while `next` comes preconfigured with one that utilizes `postcss-flexbug-fixes`.

On that note, to further help, the problem appears to be that `postcss-flexbug-fixes` removes a parenthesis, so instead of 

`flex: 1 0 calc(30% - (30px / 3));` 

it gets incorrectly transformed to:

`flex: 1 0 calc(30% - (30px / 3);`

Please file an issue [here](https://github.com/luisrudge/postcss-flexbugs-fixes).

**Comment by MrAvantiC** [comment]
Hi @mattcarlotta ,
thanks for clarifying.
I opened an issue (https://github.com/luisrudge/postcss-flexbugs-fixes/issues/68) and hope this can get fixed soon. Until then, I will just fallback to the using `next-css`. :)

Thank you!

**Comment by MrAvantiC** [comment]
Issue should be resolved, see: https://github.com/luisrudge/postcss-flexbugs-fixes/issues/68#issuecomment-620229129

Thanks for your help, guys!

**Comment by chunf** [comment]
@MrAvantiC Is there a temporary fix for now?

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
